bioblend/galaxy/tools/__init__.py

```diff
@@ -189,6 +189,41 @@ def show_tool(self, tool_id, io_details=False, link_details=False):
         params['link_details'] = link_details
         return self._get(id=tool_id, params=params)
 
+    def build(self, tool_id, inputs=None, tool_version=None, history_id=None):
+
+        """
+        Get updated tool parameters of a given tool.
+
+        :type inputs: dict
+        :param inputs: dictionary of tool parameters
+
+        :type tool_id: str
+        :param tool_id: id of the requested tool
+
+        :type history_id str
+        :param history_id: id of the requested history
+
+        :type tool_version str
+        :param tool_version: version of the requested tool
+
+        :rtype: dict
+        :return: Returns a tool model including dynamic parameters and updated values, repeats block etc.
+        """
+        params = {}
+
+        if inputs:
+            params['inputs'] = inputs
+
+        if tool_version:
+            params['tool_version'] = tool_version
+
+        if history_id:
+            params['history_id'] = history_id
+
+        url = '/'.join((self.gi.url, 'tools', tool_id, 'build'))
+
+        return self._post(payload=params, url=url)
```

Comment:
```suggestion
        return self._get(payload=params, url=url)
```
This should be a GET as far as I can see from https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/webapps/galaxy/api/tools.py#L125?

Reply:
This is possibly a doc error on the Galaxy end.

The Galaxy UI itself uses POST. I would think there might be issues using GET due to potential limitations in character lengths.

Reply:
It seems the request type is not actually specified: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/webapps/galaxy/buildapp.py#L416. A small test also reveals that POST and GET give the same result.